src/main/java/net/sacredlabyrinth/phaed/simpleclans/conversation/CreateRankNamePrompt.java

```diff
@@ -31,8 +32,14 @@
             return new MessagePromptImpl(ChatColor.RED + lang("rank.already.exists", player), this);
         }
 
-        clan.createRank(rank);
-        plugin.getStorageManager().updateClan(clan, true);
-        return new MessagePromptImpl(ChatColor.AQUA + lang("rank.created", player));
+        CreateRankEvent event = new CreateRankEvent(player, clan, clan.getRank(rank));
```

Comment:
Won't getRank return null? The rank doesn't exist yet.

Reply:
Hm, then, how I can cancel the creating rank event if the rank doesn't exist yet?
I see one way:
1) Delete rank after creating if it's canceled:
```java
clan.createRank(rank);

CreateRankEvent event = new CreateRankEvent(player, clan, clan.getRank(rank));
Bukkit.getServer().getPluginManager().callEvent(event);

if (event.isCancelled()) clan.deleteRank(rank);
```
Do you have another suggestion?

Reply:
Another way is making new event: PreCreateRankEvent, make it **Cancellable** and cancel creating the rank in `clan.createRank()` method and remove Cancellable's implementation from CreateRankEvent.

Reply:
> 
> 
> Another way is making new event: PreCreateRankEvent, make it **Cancellable** and cancel creating the rank in `clan.createRank()` method and remove Cancellable's implementation from CreateRankEvent.

I think, this way would be better.